sessions/recap-project-react/part-9/hints.md

```diff
@@ -0,0 +1,55 @@
+# Part 9 - Hints
+
+<details>
+<summary>💡 Hint: Fetching Saved Tracks</summary>
+Similar to the saved albums you need to persist a list of the savedTrackIds first and fetch the tracks later with the new endpoint.
+
+```js
+const [savedTrackIds,setSavedTrackIds] = useLocalStorageState(...)
+// ...
+const [savedTracks,setSavedTracks] = useState(...)
+
+// ...
+useEffect(() => {
+  // fetch tracks
+  setSavedTracks(receivedData)
+})
+
+```
+
+</details>
+
+<details>
+<summary>💡 Hint: Toggling saved track </summary>
+
+You need a function to toggle the saved state of each track and pass it down to the "SongList" component:
+
+```js
+function toggleSaveTrack(id) {
+  const isTrackSaved = //... check if id is in list
+  if (isTrackSaved) {
+    // remove id from list
+  } else {
+    // add id to list
+  }
+}
+```
+
+pass this function from App.js to Songlist: App => pageComponent => AlbumList => AlbumCard => SongList
+
+</details>
+
+<details>
+<summary>💡 Hint: Is Track Saved ?</summary>
+
+Your button for saving the track to the list has to know if the track is currently saved. You need to pass this information to the button from your App.js.
+
```

Comment:
Should there be a code snippet too? If not it's ok, just want to ask if something is missing here.